webapp/src/DOMJudgeBundle/Controller/API/AwardsController.php

```diff
@@ -0,0 +1,152 @@
+<?php declare(strict_types=1);
+
+namespace DOMJudgeBundle\Controller\API;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\QueryBuilder;
+use DOMJudgeBundle\Entity\Contest;
+use DOMJudgeBundle\Entity\Problem;
+use DOMJudgeBundle\Entity\Team;
+use DOMJudgeBundle\Service\DOMJudgeService;
+use DOMJudgeBundle\Service\EventLogService;
+use DOMJudgeBundle\Service\ScoreboardService;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use Swagger\Annotations as SWG;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * @Rest\Route("/api/v4/contests/{cid}/awards", defaults={ "_format" = "json" })
+ * @Rest\Prefix("/api/contests/{cid}/awards")
+ * @Rest\NamePrefix("scoreboard_")
+ * @SWG\Tag(name="Scoreboard")
+ * @SWG\Parameter(ref="#/parameters/cid")
+ * @SWG\Response(response="404", ref="#/definitions/NotFound")
+ * @SWG\Response(response="401", ref="#/definitions/Unauthorized")
+ */
+class AwardsController extends AbstractRestController
+{
+    /**
+     * @var ScoreboardService
+     */
+    protected $scoreboardService;
+
+    /**
+     * ScoreboardController constructor.
+     * @param EntityManagerInterface $entityManager
+     * @param DOMJudgeService        $DOMJudgeService
+     * @param EventLogService        $eventLogService
+     * @param ScoreboardService      $scoreboardService
+     */
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        DOMJudgeService $DOMJudgeService,
+        EventLogService $eventLogService,
+        ScoreboardService $scoreboardService
+    ) {
+        parent::__construct($entityManager, $DOMJudgeService, $eventLogService);
+        $this->scoreboardService = $scoreboardService;
+    }
+
+    /**
+     * Get the awards standings for this contest
+     * @param Request $request
+     * @return array
+     * @Rest\Get("")
+     * @SWG\Response(
+     *     response="200",
+     *     description="Returns the current teams qualifying for each award"
+     * )
+     * @throws \Exception
+     */
+    public function getAwardsAction(Request $request)
+    {
+        $public   = false;
+        if ($this->DOMJudgeService->checkrole('jury') && $request->query->has('public')) {
+            $public = $request->query->getBoolean('public');
+        }
+
+        $contest       = $this->entityManager->getRepository(Contest::class)->find($this->getContestId($request));
+        $isJury        = $this->isGranted('ROLE_JURY');
+        $accessAllowed = ($isJury && $contest->getEnabled()) || (!$isJury && $contest->isActive());
+        if (!$accessAllowed) {
+            throw new AccessDeniedHttpException();
+        }
+
+        $b = $contest->getB() ?? 0;
```

Comment:
nit: although we know what b means, I would prefer a variable name like $additionalBronzeMedals.

Reply:
Apropos medals, I was already a bit hestitant to encode their names and numers here, but I think this is the only place in DOMjudge that does something with the concept, correct?